torch/csrc/jit/codegen/cuda/ir_utils.cpp

```diff
@@ -753,6 +765,30 @@ bool isIndexSelectIndicesTv(const TensorView* tv) {
   return false;
 }
 
+bool isTorchGatherLookupTv(const Val* tv) {
+  for (auto expr : tv->uses()) {
+    if (expr->isA<TorchGatherOp>()) {
+      auto idx_sel = expr->as<TorchGatherOp>();
+      if (idx_sel->input(0) == tv) {
```

Comment:
Just curious,  when `tv->uses()` already tells that the expr is `TorchGatherOp`, it seems that there is already enough information to `return true` here? Why do we add another `idx_sel->input(0) == tv)` check here?

Reply:
Aha, got it, it is used to distinguish between different argument of `TorchGather`, ignore pls.